eoc_journal/eoc_journal.py

```diff
@@ -97,6 +110,44 @@ def student_view(self, context=None):
         )
         return fragment
 
+    def list_user_pb_answers_by_section(self, user_id, course_id):
+        """
+        Returns a list of dicts with pb-answers grouped by section.
+        """
+        # Get the selected blocks and their answers
+        blocks = self.list_pb_answers()
+        answers_names = [block['name'] for block in blocks]
+        course_key = CourseKey.from_string(course_id)
```

Comment:
If we do [what problem-builder does](https://github.com/open-craft/problem-builder/blob/27e2f4ba5afdefcb6a4e5fd793969e1e626a0abe/problem_builder/answer.py#L56-L58), then we don't have to convert the key to a string, and then back to `CourseKey` again.